File: Preprocessing_data-price_of_house.R
df <- read.csv("D:/PROJEKTY/MachineLearning/ANN-tutorial/Data Files/House_Price.csv", header = TRUE)

summary(df)
hist(df$crime_rate)
pairs(~price+crime_rate+n_hot_rooms+rainfall,data = df)
barplot(table(df$waterbody))

# Outlier treatment

uv = 3*quantile(df$n_hot_rooms, 0.99)
df$n_hot_rooms[df$n_hot_rooms > uv ] <- uv
lv = 0.3 * quantile(df$rainfall, 0.01)
df$rainfall[df$rainfall < lv] <- lv

# Missing value

df$n_hos_beds[is.na(df$n_hos_beds)] <- mean(df$n_hos_beds, na.rm = TRUE)


# Variable transformation

plot(df$price,df$crime_rate)

df$crime_rate = log(1+df$crime_rate)

df$avg_dist = (df$dist1 + df$dist2 + df$dist3 + df$dist4)/4

df2 <- df[,-7:-10]
df <- df2
rm(df2)

df <- df[,-14]


# DUMMY variables

#install.packages("dummies")
library(dummies)

df <- dummy.data.frame(df)
df <- df[,-9]
df <- df[,-14]

File: Preprocessing_data-price_of_house.py
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
import seaborn as sns 

df = pd.read_csv("./Data Files/House_Price.csv", header=0)

# Outlier treatment

uv = np.percentile(df.n_hot_rooms, [99])[0]
df.n_hot_rooms[(df.n_hot_rooms > 3 * uv)] = 3 * uv
#print(data_frame.n_hot_rooms[(data_frame.n_hot_rooms > uv)])

lv = np.percentile(df.rainfall, [1])[0]
df.rainfall[(df.rainfall < 0.3 * lv)] = 0.3 * lv

# Missing value

df.n_hos_beds = df.n_hos_beds.fillna(df.n_hos_beds.mean())

# Variable transformation

#sns.jointplot(x = "crime_rate", y = "price", data = df)
#plt.show()

df.crime_rate = np.log(1+df.crime_rate)
#sns.jointplot(x = "crime_rate", y = "price", data = df)
#plt.show() - more linear

df["avg_dist"] = (df.dist1 + df.dist2 + df.dist3 + df.dist4)/4
del df["dist1"]
del df["dist2"]
del df["dist3"]
del df["dist4"]

del df["bus_ter"]

# Dummy variable

df = pd.get_dummies(df)
#print(df.head())
del df["airport_NO"]
del df["waterbody_None"]
#print(df.head())

# Test-train

from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split